apps/aecore/src/aec_tx_pool.erl

```diff
@@ -375,6 +384,38 @@ handle_call_({delete, TxHash}, _From, #state{dbs = Dbs} = State) ->
             none -> {error, not_found}
         end,
     {reply, Res, State};
+handle_call_({failed_txs, FailedTxs}, _From, #state{dbs = Dbs} = State) ->
+    #dbs{db = Db, visited_db = VisitedDb} = Dbs,
+    TryUpdating =
+        fun(DBHandle, Key, SignedTx, FailReason) ->
+            case ets:lookup(DBHandle, Key) of
+                [{Key, #tx{failures = Failures0} = TxRec0}] ->
+                    Failures = Failures0 + 1,
+                    case should_delete_tx(SignedTx,
+                                          FailReason,
+                                          Failures) of
+                        true ->
+                            lager:debug("Tx reached ~p failures (failed with: ~p)", [Failures, FailReason]),
+                            pool_db_raw_delete(Dbs, Key),
```

Comment:
Is this sufficient? Shouldn't we also call `aec_db:remove_tx_from_mempool(TxHash)`?